### PR TITLE
Downgrade gulp-mocha to v3 to fix server tests

### LINKED ISSUE #2583: Server tests fail due to improperly initialized mongoose connection
When `npm run test:server` is run, the tests fail with
```
Assertion failed: You must provide either mongoUrl|clientPromise|client in options
```

This happens because `mongoose` that we import to server tests, doesn't have `mongoose.connection.client`.

In gulp, everything gets properly initialized, so it seems like `mongoose` imported into tests is different from `mongoose` that is set up with gulp...

For the same reason, server tests fail in github workflows here.

To be continued...

### PR DESCRIPTION
**Pull request by mrkvon**
### What

Downgrade `gulp-mocha`. #2461 added a breaking change in `gulp-mocha` that broke our server tests.

Also, fix a few tests that got broken in the meantime.

Fixes #2583
~~(Hopefully 🤞🏾)~~ _update: yes it does :tada:_

### Why

Related to https://github.com/sindresorhus/gulp-mocha/issues/166. Since `gulp-mocha v4`, apparently mocha runs in its own process, separate from gulp. Therefore communication between gulpfile (which sets up mongoose connection) and server tests (which need mongoose connection) is broken.

### To Do Later

As the mentioned issue above suggests (https://github.com/sindresorhus/gulp-mocha/issues/166#issuecomment-285553304), we shouldn't have mixed test logic and gulp in the first place.
To resolve this cleanly, we should fix our tests, and then upgrade gulp-mocha, or even better, stop running tests with gulp in the first place (#1149)

### CODE REVIEW

**Comment by mrkvon** [comment]
Unfortunately, workflows fail for me at the moment. Trying to resolve.

**Comment by mrkvon** [comment]
@sitarane 

seems like we have the tests passing. but it's a dirty fix. our tests will get broken again as soon as we upgrade `gulp-mocha` again above version 3. They need to be fixed, and gulp-independent (or not use gulp at all...)

**Review by sitarane — APPROVED** [review]
I fixed another test. Maybe have a look before merging.

File: package-lock.json [hunk 1]
```diff
@@ -7562,6 +7562,12 @@
       "integrity": "sha512-9o5UecI3GhkpM6DrXr69PblIuWxPKk9Y0jHBRhdocZ2y7YECBFCsHm79Pr3OyR2AvjhDkabFJaDJMYRazHgsow==",
```
[the remark this answers is not in the document]
Comment (mrkvon) [reply package-lock.json:7562]: It was generated automatically when i ran `npm install gulp-mocha@3`. I guess all are dependency changes of gulp-mocha, but i don't really see behind the scenes of this...

**Pull request merged**